go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -143,56 +143,8 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 	var oldMasterTablet *topodatapb.Tablet
 	oldMasterAlias := si.MasterAlias
 
-	// Update the tablet records concurrently.
-	event.DispatchUpdate(ev, "updating old and new master tablet records")
-	log.Infof("finalizeTabletExternallyReparented: updating tablet records")
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		// Update our own record to master.
-		_, err := agent.TopoServer.UpdateTabletFields(ctx, agent.TabletAlias,
-			func(tablet *topodatapb.Tablet) error {
-				tablet.Type = topodatapb.TabletType_MASTER
-				return nil
-			})
-		if err != nil {
-			errs.RecordError(err)
-		}
-	}()
-
-	if !topoproto.TabletAliasIsZero(oldMasterAlias) {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-
-			// Forcibly demote the old master in topology, since we can't rely on the
-			// old master to be up to change its own record.
-			var err error
-			oldMasterTablet, err = agent.TopoServer.UpdateTabletFields(ctx, oldMasterAlias,
-				func(tablet *topodatapb.Tablet) error {
-					tablet.Type = topodatapb.TabletType_REPLICA
-					return nil
-				})
-			if err != nil {
-				errs.RecordError(err)
-				return
-			}
-
-			// We now know more about the old master, so add it to event data.
-			ev.OldMaster = *oldMasterTablet
-		}()
-	}
-
+	// update the shard record first
```

Comment:
Why update the shard record first? Traditionally, updating the shard record means the reparent is done. That's how we ensure that subsequent TER calls will fix the tablet records if this attempt fails.

Reply:
take a look at the latest and let me know if you still think this is a problem